### PR TITLE
update docs for functions supporting COST and DEFAULT arg clauses

### PR DESCRIPTION
**Pull request by lisakowen**
updates include:
- added COST and DEFAULT argument clauses and associated discussions to the CREATE FUNCTION page
- added COST clause and discussion to ALTER FUNCTION page
- added a "using default arguments" example to the PL/pgSQL page

question: does gporca use the execution_cost for anything, or is it just the planner that uses the info?

### CODE REVIEW

**Comment by vraghavan78** [comment]
execution cost is used by planner and GPORCA in picking the cheapest plan. Besides these two components, execution cost is primarily used by resource manager. @foyzur @yydzero correct me if  I am wrong here.

File: gpdb-doc/dita/ref_guide/extensions/pl_sql.xml [hunk 1]
```diff
@@ -252,6 +252,51 @@ $$ LANGUAGE plpgsql VOLATILE;</codeblock>
           inside a PL/pgSQL function, use the SQL command <codeph>CREATE TABLE AS</codeph>.</note>
       </body>
     </topic>
+
+    <topic xml:lang="en" id="topic_lsh_5n5_2z1313">
+      <title>Example: Using Default Argument Values</title>
+      <body>
+        <p>You can declare PL/pgSQL functions with default values for some or all input arguments.
+          The default values are inserted whenever the function is called with insufficiently many
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/extensions/pl_sql.xml]: Trying to figure out a clearer wording for "insufficiently many."  Maybe "fewer than the declared number of" ?

File: gpdb-doc/dita/ref_guide/extensions/pl_sql.xml [hunk 2]
```diff
@@ -252,6 +252,51 @@ $$ LANGUAGE plpgsql VOLATILE;</codeblock>
...
+          actual arguments. Since arguments can only be omitted from the end of the actual argument
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/extensions/pl_sql.xml]: Since -> Because

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml [hunk 3]
```diff
@@ -114,6 +121,14 @@ to be executed with the privileges of the user that created it. The key
 word <codeph>EXTERNAL</codeph> is allowed for SQL conformance, but it
 is optional since, unlike in SQL, this feature applies to all functions
 not just external ones.</pd></plentry>
+<plentry> <pt>COST <varname>execution_cost</varname></pt>
+<pd>A positive number identifying the estimated execution cost for the function,
+in <xref href="https://www.postgresql.org/docs/8.3/static/runtime-config-query.html#GUC-CPU-OPERATOR-COST" scope="external" format="html">cpu_operator_cost</xref> units.
+If the function returns a set, <varname>execution_cost</varname> identifies the
+cost per returned row. If the cost is not specified, C-language and internal
+functions default to 1 unit, while 100 units is the default for functions in
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml]: Edit to "while functions in all other languages default to 100 units."

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml [hunk 4]
```diff
@@ -114,6 +121,14 @@ to be executed with the privileges of the user that created it. The key
...
+all other languages. Larger values cause the planner to try to avoid evaluating
+the function more often than necessary.</pd></plentry>
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml]: This wording is a little strange, as it implies that by default the planner tries to evaluate the function more often than is necessary :-)  Maybe just make it "With larger values, the planner tries to evaluate the function less often than with smaller values."?

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml [hunk 5]
```diff
@@ -153,9 +168,16 @@ be used for several different functions so long as they have distinct
 argument types. However, the C names of all functions must be different,
 so you must give overloaded C functions different C names (for example,
 use the argument types as part of the C names). </p><p>Two functions are considered the same if they have the same names
-and input argument types, ignoring any OUT parameters. Thus for example
+and input argument types, ignoring any <codeph>OUT</codeph> parameters. Thus for example
 these declarations conflict:</p><codeblock>CREATE FUNCTION foo(int) ...
-CREATE FUNCTION foo(int, out text) ...</codeblock><p>When repeated <codeph>CREATE FUNCTION</codeph> calls refer to the same
+CREATE FUNCTION foo(int, out text) ...</codeblock>
+<p>Functions that have different argument type lists will not be considered to conflict at
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml]: "will not be" -> "are not"

**Pull request merged**